controller_manager:
  ros__parameters:
    update_rate: 100  # Hz
    
    # List of controllers to be loaded
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    
    # Swerve controller (replaces separate steer/drive controllers)
    swerve_controller:
      type: rse_shl1_swerve_controller/SwerveController


# Joint State Broadcaster - publishes all joint states to /joint_states
joint_state_broadcaster:
  ros__parameters:
    joints:
      - fl_steer_joint
      - fl_drive_joint
      - fr_steer_joint
      - fr_drive_joint
      - ml_steer_joint
      - ml_drive_joint
      - mr_steer_joint
      - mr_drive_joint
      - rl_steer_joint
      - rl_drive_joint
      - rr_steer_joint
      - rr_drive_joint
    
# Swerve Controller Configuration
swerve_controller:
  ros__parameters:
    # Robot geometry (from URDF)
    wheel_radius: 0.0825
    wheelbase: 0.54
    track_width: 0.53
    
    # Joint names in order: FL, FR, ML, MR, RL, RR
    steer_joints:
      - fl_steer_joint
      - fr_steer_joint
      - ml_steer_joint
      - mr_steer_joint
      - rl_steer_joint
      - rr_steer_joint
    
    drive_joints:
      - fl_drive_joint
      - fr_drive_joint
      - ml_drive_joint
      - mr_drive_joint
      - rl_drive_joint
      - rr_drive_joint